radio/src/gui/480x272/lcd.cpp

```diff
@@ -557,3 +557,272 @@ void DMABitmapConvert(uint16_t * dest, const uint8_t * src, uint16_t w, uint16_t
+  }
+  while (x < y) {
+    if (f >= 0) {
+      y--;
+      ddF_y += 2;
+      f += ddF_y;
+    }
+    x++;
+    ddF_x += 2;
+    f += ddF_x;
+    if (x < (y + 1)) {
+      if (corners & 0x01) {
+    	lcd->drawSolidVerticalLine(x0 + x, y0 - y, 2 * y + 1, att);
+      }
+      if (corners & 0x02) {
+    	lcd->drawSolidVerticalLine(x0 - x, y0 - y, 2 * y + 1, att);
+      }
+    }
+    if (y != py) {
+      if (corners & 0x01) {
+    	lcd->drawSolidVerticalLine(x0 + py, y0 - px, 2 * px + 1, att);
+      }
+      if (corners & 0x02) {
+    	lcd->drawSolidVerticalLine(x0 - py, y0 - px, 2 * px + 1, att);
+      }
+      py = y;
+    }
+    px = x;
+  }
+}
+
+void lcdFillTriangle(coord_t x0, coord_t y0, coord_t x1, coord_t y1, coord_t x2, coord_t y2, LcdFlags att)
+{
+  int32_t a, b;
+
+  #define SWAP(xx,yy) {int32_t t = xx; xx = yy; yy = t;}
+
+  if (y0 > y1) { SWAP(y0, y1); SWAP(x0, x1); }
+  if (y1 > y2) { SWAP(y2, y1); SWAP(x2, x1); }
+  if (y0 > y1) { SWAP(y0, y1); SWAP(x0, x1); }
+
+  if (y0 == y2) { // Handle awkward all-on-same-line case as its own thing
+    a = b = x0;
+    if (x1 < a)
+      a = x1;
+    else if (x1 > b)
+      b = x1;
+    if (x2 < a)
+      a = x2;
+    else if (x2 > b)
+      b = x2;
+    lcd->drawSolidHorizontalLine(a, y0, b - a + 1, att);
+    return;
+  }
+
+  int32_t dx01 = x1 - x0, dy01 = y1 - y0, dx02 = x2 - x0, dy02 = y2 - y0, dx12 = x2 - x1, dy12 = y2 - y1;
+  int32_t sa = 0, sb = 0;
+  int32_t last = (y1 == y2) ? y1 : y1 - 1;
+  int32_t y;
+
+  for (y = y0; y <= last; y++) {
+    a = x0 + sa / dy01;
+    b = x0 + sb / dy02;
+    sa += dx01;
+    sb += dx02;
+    if (a > b) SWAP(a, b);
+    lcd->drawSolidHorizontalLine(a, y, b - a + 1, att);
+  }
+
+  sa = dx12 * (y - y1);
+  sb = dx02 * (y - y0);
+
+  for (; y <= y2; y++) {
+    a = x1 + sa / dy12;
+    b = x0 + sb / dy02;
+    sa += dx12;
+    sb += dx02;
+    if (a > b) SWAP(a, b);
+    lcd->drawSolidHorizontalLine(a, y, b - a + 1, att);
+  }
+}
+
+// Cohen�Sutherland, see wikipedia
+
+// Compute the bit code for a point (x, y) using the clip rectangle
+// bounded diagonally by (xmin, ymin), and (xmax, ymax)
+uint8_t ComputeOutCode(float x, float y, float xmin, float xmax, float ymin, float ymax)
+{
+  uint8_t code = 0;
+  if (x < xmin)
+	code |= 1;
+  else if (x > xmax)
+	code |= 2;
+  if (y < ymin)
+	code |= 8;
+  else if (y > ymax)
+	code |= 4;
+  return code;
+}
+
+// clipping algorithm clips a line from P0 = (x0, y0) to P1 = (x1, y1) against a rectangle with
+// diagonal from (xmin, ymin) to (xmax, ymax).
+void CohenSutherlandLineClipAndDraw(float x0, float y0, float x1, float y1, float xmin, float xmax, float ymin, float ymax, uint8_t pat=SOLID, LcdFlags att=0)
+{
+  uint8_t outcode0 = ComputeOutCode(x0, y0, xmin, xmax, ymin, ymax);
+  uint8_t outcode1 = ComputeOutCode(x1, y1, xmin, xmax, ymin, ymax);
+  bool accept = false;
+
+  while (true) {
+    if (!(outcode0 | outcode1)) {
+      accept = true;
+      break;
+    } else if (outcode0 & outcode1) {
```

Comment:
wrong code style

Reply:
could you pl give me a hint? I don't see it.

Reply:
We put the else on the next line after the closing brace

Reply:
ah. thx. Done.
it's a bit difficult with opentx since it appears to follow various styles in the different files; maybe some clang or so could be worthwhile